player/lua/utils.lua

```diff
@@ -0,0 +1,84 @@
+--[[
+This file is part of mpv.
+
+mpv is free software; you can redistribute it and/or
+modify it under the terms of the GNU Lesser General Public
+License as published by the Free Software Foundation; either
+version 2.1 of the License, or (at your option) any later version.
+
+mpv is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public
+License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+]]
+
+local utils = require "mp.utils"
+
+local default_config_file = [[# https://mpv.io/manual/master/#configuration-files
+# https://mpv.io/manual/master/#options
+
+]]
+
+local default_input_conf = [[# https://mpv.io/manual/master/#command-interface
+
+]]
+
+local function show_error(message)
+    mp.msg.error(message)
+    if mp.get_property_native("vo-configured") then
+        mp.osd_message(message)
+    end
+end
+
+local function edit_config_file(filename, initial_contents)
+    local path = mp.find_config_file(filename)
+
+    if not path then
+        path = mp.command_native({"expand-path", "~~/" .. filename})
+        local file_handle, error_message = io.open(path, "w")
+
+        if not file_handle then
+            show_error(error_message)
+            return
+        end
+
+        file_handle:write(initial_contents .. "\n")
+        file_handle:close()
+    end
+
+    local platform = mp.get_property("platform")
+    local args
+    if platform == "windows" then
+        -- Trying to open a path with a comma fails unless there's a space in
+        -- the path.
+        args = {"explorer", mp.command_native({"normalize-path", path}) .. " "}
```

Comment:
Another issue with `explorer.exe` is that it seems to return `1` always. which triggers `show_error` below. Maybe instead of hacking explorer.exe with space at the end it is better to call url handler directly. This also doesn't need normalize.

```suggestion
        args = {"rundll32", "url.dll,FileProtocolHandler", path}
```